lib/instream/query/builder.ex

```diff
@@ -131,6 +131,9 @@ defmodule Instream.Query.Builder do
   Builds a `WHERE` query expression.
   """
   @spec where(t, map) :: t
+  def where(query, nil   ), do: query
+  def where(query, []    ), do: query
+  def where(query, fields) when fields == %{}, do: query
```

Comment:
I'm in favor of this change, as it's consistent with the way query building works in other languages. The use case I'm thinking of is that of delegating query creation to code with different responsibilities. If one simply chooses not to alter the query, this allows them to do so.  

If the way it works now is simply to clear what was there before, then there's no possibility of composition.